src/hockeypuck/pghkp/storage.go

```diff
@@ -91,6 +91,157 @@ var drConstraintsSQL = []string{
 	`DROP INDEX subkeys_rfp;`,
 }
 
+var crTempTablesSQL = []string{
+	`CREATE TEMPORARY TABLE IF NOT EXISTS keys_copyin (
+rfingerprint TEXT,
+doc jsonb,
+ctime TIMESTAMP WITH TIME ZONE,
+mtime TIMESTAMP WITH TIME ZONE,
```

Comment:
Just curious, why not add NOT NULL constraints on these? I think it might simplify some of the queries below.

Reply:
This would mean the initial insert may return an error, if a field I have sent to the DB is NULL. Then I would have to treat this as a reason to stop bulk processing and revert to one-key-at-a-time.

Either I check the strings, before I send them to the DB, or I let the DB check them...